dashboard/client/src/components/FunctionTable/FunctionTable.jsx

```diff
@@ -1,71 +1,149 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { Table } from 'reactstrap';
+import { Table, UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
 
 import { withRouter } from 'react-router-dom';
 
-import { FunctionTableItem } from '../FunctionTableItem';
+import { FunctionTableItem, genOwnerInitials } from '../FunctionTableItem';
 
 import './FunctionTable.css'
 
-function renderBody(fns, user, clickHandler) {
+function renderBody(fns, user, clickHandler, filter) {
   if (fns.length === 0) {
     return (
       <tr>
         <td>No functions available.</td>
       </tr>
     );
   } else {
-    return fns.map((fn) => {
+    return fns.filter(({ gitOwner }) => filter(gitOwner)).map((fn) => {
+
       return (
-        <FunctionTableItem key={fn.shortSha} fn={fn} user={user} onClick={clickHandler} />
+        <FunctionTableItem key={fn.shortSha + fn.name} fn={fn} user={user} onClick={clickHandler} />
       );
     });
   }
 }
 
-const FunctionTable = withRouter(({ isLoading, fns, user, history }) => {
-  const onRowClick = to => history.push(to);
-  const tbody = isLoading ? (
-    <tr>
-      <td colSpan="8" style={{ textAlign: 'center' }}>
-        <FontAwesomeIcon icon="spinner" spin />
-      </td>
-    </tr>
-  ) : (
-    renderBody(fns, user, onRowClick)
-  );
-
-  let tableProps = {
-    responsive: true,
-    className: 'function-table',
-  };
-
-  if (fns && fns.length > 0) {
-    tableProps.hover = true;
+class FunctionTableWithoutRouter extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filter: localStorage.getItem('filter') || '',
```

Comment:
Should we document this somewhere in the dashboard's README? User privacy guide etc?